packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -865,6 +910,47 @@ export class Toolkit extends CloudAssemblySourceBuilder {
     return ret;
   }
 
+  /**
+   * Refactor Action. Moves resources from one location (stack + logical ID) to another.
+   */
+  public async refactor(cx: ICloudAssemblySource, options: RefactorOptions = {}): Promise<void> {
```

Comment:
Refactor should probably return some useful structure as well, shouldn't it?

Reply:
Possibly. I'm confused as to what should get notified to the `IoHost`, and what should be returned.